gms/gossiper.cc

```diff
@@ -1633,7 +1627,7 @@ future<> gossiper::real_mark_alive(inet_address addr) {
 
 future<> gossiper::mark_dead(inet_address addr, endpoint_state& local_state, permit_id pid) {
     logger.trace("marking as down {}", addr);
-    local_state.mark_dead();
+    auto permit = co_await lock_endpoint(addr, pid);
```

Comment:
A sneaked-in bugfix? Should probably be mentioned in commit message.

Reply:
hmm, good catch.
It somehow escaped https://github.com/scylladb/scylladb/pull/14845
I submit it as a separate patch, or even a separate PR.